native/core/src/common/bit.rs

```diff
@@ -671,7 +671,7 @@ impl BitReader {
             }
         }
 
-        debug_assert!((offset + i) % 8 != 0 || num_bits_to_read - i < 8);
+        debug_assert!((offset + i).is_multiple_of(8) || num_bits_to_read - i < 8);
```

Comment:
This seems to be inverted?

```suggestion
        debug_assert!(!(offset + i).is_multiple_of(8) || num_bits_to_read - i < 8);
```